gemini/sample-apps/genai-mlops-tune-and-eval/local/pipeline.py

```diff
@@ -0,0 +1,253 @@
+        )
+        qa_metrics = pointwise_metric.metrics_table[
+            [
+                col
+                for col in pointwise_metric.metrics_table.columns
+                if ("question_answering" in col) or ("groundedness" in col)
+            ]
+        ].to_dict("records")[0]
+
+        return pairwise_best_response, qa_metrics
+
+    # Compare response from baseline model to candidate model to see which is better
+    baseline_model = GenerativeModel(
+        baseline_model_endpoint,
+        generation_config={
+            "temperature": 0.4,
+            "max_output_tokens": 512,
+        },
+    )
+    candidate_model = GenerativeModel(
+        candidate_model_endpoint,
+        generation_config={
+            "temperature": 0.4,
+            "max_output_tokens": 512,
+        },
+    )
+
+    instruction_qa = "Analyze the glucose trends in the glucose values provided in the CSV contained in the context. Ensure the analysis you provide can easily be understood by a diabetes patient with no medical expertise."
+    context_qa = (
+        "Context:\n"
+        + "```csv\ndate,time,patient ID,glucose\n2024-11-12,7:00 AM,1,80\n2024-11-12,8:00 AM,1,96\n2024-11-12,11:00 AM,1,90\n2024-11-12,12:00 PM,1,115\n2024-11-12,5:00 PM,1,77\n2024-11-12,6:00 PM,1,80\n2024-11-13,7:00 AM,1,94\n2024-11-13,8:00 AM,1,100\n2024-11-13,11:00 AM,1,87\n2024-11-13,12:00 PM,1,126\n2024-11-13,5:00 PM,1,71\n2024-11-13,6:00 PM,1,82\n2024-11-14,7:00 AM,1,84\n2024-11-14,8:00 AM,1,72\n2024-11-14,11:00 AM,1,96\n2024-11-14,12:00 PM,1,110\n2024-11-14,5:00 PM,1,99\n2024-11-14,6:00 PM,1,74\n2024-11-15,7:00 AM,1,96\n2024-11-15,8:00 AM,1,97\n2024-11-15,11:00 AM,1,99\n2024-11-15,12:00 PM,1,130\n2024-11-15,5:00 PM,1,99\n2024-11-15,6:00 PM,1,87\n2024-11-16,7:00 AM,1,89\n2024-11-16,8:00 AM,1,92\n2024-11-16,11:00 AM,1,77\n2024-11-16,12:00 PM,1,105\n2024-11-16,5:00 PM,1,79\n2024-11-16,6:00 PM,1,90\n2024-11-17,7:00 AM,1,74\n2024-11-17,8:00 AM,1,82\n2024-11-17,11:00 AM,1,74\n2024-11-17,12:00 PM,1,78\n2024-11-17,5:00 PM,1,95\n2024-11-17,6:00 PM,1,74\n2024-11-18,7:00 AM,1,95\n2024-11-18,8:00 AM,1,87\n2024-11-18,11:00 AM,1,79\n2024-11-18,12:00 PM,1,90\n2024-11-18,5:00 PM,1,79\n2024-11-18,6:00 PM,1,77\n"
+    )
+    prompt_qa = instruction_qa + "\n" + context_qa + "\n\nAnswer:\n"
+
+    baseline_model_response = baseline_model.generate_content(
+        contents=prompt_qa,
+    )
+    candidate_model_response = candidate_model.generate_content(
+        contents=prompt_qa,
+    )
+    responses = [
+        baseline_model_response.candidates[0].text,
+        candidate_model_response.candidates[0].text,
+    ]
+
+    best_response, metrics = rank_responses(instruction_qa, context_qa, responses)
+
+    for ix, response in enumerate(responses, start=1):
+        print(f"Response no. {ix}: \n {response}")
+
+    print(best_response)
+    # metrics
+
+
+@dsl.pipeline
+def gemini_tuning_pipeline(
+    project: str = "genai-mlops-tune-and-eval",
```

Comment:
Change this to a placeholder
```suggestion
    project: str = "YOUR_PROJECT_ID",
```